packages/mgt/src/components/mgt-get/mgt-get.ts

```diff
@@ -219,40 +254,57 @@ export class MgtGet extends MgtTemplatedComponent {
           request = request.middlewareOptions(prepScopes(...this.scopes));
         }
 
-        let response = await request.get();
+        let response = null;
+        if (this.type == ResponseType.json) {
+          response = await request.get();
 
-        if (isDeltaLink && this.response && Array.isArray(this.response.value) && Array.isArray(response.value)) {
-          response.value = this.response.value.concat(response.value);
-        }
+          if (isDeltaLink && this.response && Array.isArray(this.response.value) && Array.isArray(response.value)) {
+            response.value = this.response.value.concat(response.value);
+          }
 
-        if (!this.isPolling && !equals(this.response, response)) {
-          this.response = response;
-        }
+          if (!this.isPolling && !equals(this.response, response)) {
+            this.response = response;
+          }
+
+          // get more pages if there are available
+          if (response && Array.isArray(response.value) && response['@odata.nextLink']) {
+            let pageCount = 1;
+            let page = response;
 
-        // get more pages if there are available
-        if (response && Array.isArray(response.value) && response['@odata.nextLink']) {
-          let pageCount = 1;
-          let page = response;
-
-          while (
-            (pageCount < this.maxPages || this.maxPages <= 0 || (isDeltaLink && this.pollingRate)) &&
-            page &&
-            page['@odata.nextLink']
-          ) {
-            pageCount++;
-            const nextResource = page['@odata.nextLink'].split(this.version)[1];
-            page = await graph.client
-              .api(nextResource)
-              .version(this.version)
-              .get();
-            if (page && page.value && page.value.length) {
-              page.value = response.value.concat(page.value);
-              response = page;
-              if (!this.isPolling) {
-                this.response = response;
+            while (
+              (pageCount < this.maxPages || this.maxPages <= 0 || (isDeltaLink && this.pollingRate)) &&
+              page &&
+              page['@odata.nextLink']
+            ) {
+              pageCount++;
+              const nextResource = page['@odata.nextLink'].split(this.version)[1];
+              page = await graph.client
+                .api(nextResource)
+                .version(this.version)
+                .get();
+              if (page && page.value && page.value.length) {
+                page.value = response.value.concat(page.value);
+                response = page;
+                if (!this.isPolling) {
+                  this.response = response;
+                }
               }
             }
           }
+        } else {
+          if (this.resource.indexOf('/photo/$value') == -1) {
+            throw new Error('Only /photo/$value endpoints support the photo type');
```

Comment:
For added clarity, I suggest changing `photo` to `image` to match the type

```suggestion
            throw new Error('Only /photo/$value endpoints support the image type');
```